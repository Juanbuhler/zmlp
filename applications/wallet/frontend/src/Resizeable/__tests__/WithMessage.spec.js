import TestRenderer from 'react-test-renderer'

import ResizeableWithMessage from '../WithMessage'

describe('<Resizeable />', () => {
  it('should render properly initially closed', () => {
    const component = TestRenderer.create(
      <ResizeableWithMessage
        storageName="CursorLeftWithMessage"
        minSize={400}
        openToThe="left"
        header={() => <div>Header</div>}
        isInitiallyOpen={false}
      >
        {() => <div>Body</div>}
      </ResizeableWithMessage>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly with the cursor left', () => {
    const component = TestRenderer.create(
      <ResizeableWithMessage
        storageName="CursorLeftWithMessage"
        minSize={400}
        openToThe="left"
        header={() => <div>Header</div>}
        isInitiallyOpen
      >
        {() => <div>Body</div>}
      </ResizeableWithMessage>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
