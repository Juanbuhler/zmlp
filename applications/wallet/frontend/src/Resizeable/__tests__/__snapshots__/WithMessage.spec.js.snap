// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resizeable /> should render properly initially closed 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: col-resize;
  width: 4px;
  margin-left: -2px;
  margin-right: -2px;
  z-index: 10;
}

.emotion-1 {
  width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "height": "100%",
    }
  }
>
  <div
    aria-label="Resize horizontally"
    className="emotion-0"
    onMouseDown={[Function]}
  />
  <div>
    Header
  </div>
  <div
    className="emotion-1"
  >
    <div>
      Body
    </div>
  </div>
</div>
`;

exports[`<Resizeable /> should render properly with the cursor left 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: col-resize;
  width: 4px;
  margin-left: -2px;
  margin-right: -2px;
  z-index: 10;
}

.emotion-1 {
  width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "height": "100%",
    }
  }
>
  <div
    aria-label="Resize horizontally"
    className="emotion-0"
    onMouseDown={[Function]}
  />
  <div>
    Header
  </div>
  <div
    className="emotion-1"
  >
    <div>
      Body
    </div>
  </div>
</div>
`;
